src/lib/viz/dataview/mode/DataViewMode.ts

```diff
@@ -14,6 +14,12 @@ export abstract class DataViewMode extends WithEvents {
 
     this.column = column;
     this.dataSource = dataSource;
+
+    if (this.dataSource instanceof Layer) {
+      this.dataSource.addSourceField(this.column);
```

Comment:
I would suggest renaming dataSource to something else, like 'origin'. Everytime I read dataSource I just can't avoid thinking about just 'Source' and not about 'Layer'.

Reply:
renamed to `dataOrigin`